From: Andrii Nakryiko <andrii@kernel.org>
Subject: exit: document sched_process_exit and sched_process_template relation
Date: Thu, 3 Apr 2025 10:41:20 -0700

Add an explicit note pointing out that sched_process_exit tracepoint is,
conceptually, related to sched_process_template and should be kept in sync
(though, on the best effort basis).

Link: https://lkml.kernel.org/r/20250403174120.4087794-1-andrii@kernel.org
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Suggested-by: Ingo Molnar <mingo@kernel.org>
Cc: Alexander Potapenko <glider@google.com>
Cc: Christian Brauner <brauner@kernel.org>
Cc: "Masami Hiramatsu (Google)" <mhiramat@kernel.org>
Cc: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Cc: Michal Hocko <mhocko@kernel.org>
Cc: Oleg Nesterov <oleg@redhat.com>
Cc: Steven Rostedt (Google) <rostedt@goodmis.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/trace/events/sched.h |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/include/trace/events/sched.h~exit-move-and-extend-sched_process_exit-tracepoint-fix
+++ a/include/trace/events/sched.h
@@ -326,7 +326,11 @@ DEFINE_EVENT(sched_process_template, sch
 	     TP_ARGS(p));
 
 /*
- * Tracepoint for a task exiting:
+ * Tracepoint for a task exiting.
+ * Note, it's a superset of sched_process_template and should be kept
+ * compatible as much as possible. sched_process_exits has an extra
+ * `group_dead` argument, so sched_process_template can't be used,
+ * unfortunately, just like sched_migrate_task above.
  */
 TRACE_EVENT(sched_process_exit,
 
_
